rpc/client/interface.go

```diff
@@ -24,17 +24,27 @@ import (
 	"context"
 
 	"github.com/tendermint/tendermint/libs/bytes"
-	"github.com/tendermint/tendermint/libs/service"
 	ctypes "github.com/tendermint/tendermint/rpc/core/types"
 	"github.com/tendermint/tendermint/types"
 )
 
 //go:generate ../../scripts/mockery_generate.sh Client
 
-// Client wraps most important rpc calls a client would make if you want to
-// listen for events, test if it also implements events.EventSwitch.
+// Client is the interface that must be satisfied by implementations of a
+// client of the Tendermint RPC service.
 type Client interface {
-	service.Service
+	// These methods define the operational structure of the client.
+
+	// Start the client. Start must report an error if the client is running.
+	Start() error
+
+	// Stop the client. Stop must report an error if the client is not running.
+	Stop() error
+
+	// IsRunning reports whether the client is running.
+	IsRunning() bool
```

Comment:
I completely agree. 

1. Some clients do need to initialize and tear down, but ideally that should happen during construction. Clients should probably have a Close method that users are required to call, to give them an opportunity to clean up. Here, I've mostly hewed to the existing code, so I wouldn't have to make more infectious changes until after the dependency on service is broken.
2. The Service interface is a monstrosity,  and I would also like to do some cleanup there. Right now I'm focused on RPC, but one of the benefits of decoupling the two is that it will become easier to make changes on both sides without them pulling each other.

Reply:
My thought here likely comes from the fact that I think having a `service.Service` type as a unifying abstraction doesn't necessarily make sense and a lot of the headache we have around it is trying to force a lot of 'things that need to start running' to look like a `service.Service`. To that point, I think that clients do not need to look like `service.Service` almost at all so it should be pretty safe to safe to start down the path of removing this coupling completely.

Perhaps you haven't gotten to this point in the design yet, but I'm wondering if you have an idea for how to handle websockets that disconnect outside of a method call and therefore don't quickly report the code using them?

Reply:
> I think that clients do not need to look like `service.Service` almost at all so it should be pretty safe to safe to start down the path of removing this coupling completely.

Yes, agreed. I would like to make some more substantive changes to the way RPCs are implemented, but some of that will probably want an ADR (or maybe more than one) for interface changes. I'm starting with relatively low-cost prep work to make "real" changes easier to land without blowing things up.

> Perhaps you haven't gotten to this point in the design yet, but I'm wondering if you have an idea for how to handle websockets that disconnect outside of a method call and therefore don't quickly report the code using them?

I have some ideas about how to make it better, but I am going to have to experiment with it. I think probably the right solution is to record a connection failure in the client and report it back through the calls (or similar).

Reply:
Leaving this thread here for posterity.